### PR TITLE
skarnet skalibs, execline and s6-* version bump

### PR DESCRIPTION
**Pull request by arkamar**
This is a version bump of `dev-libs/skalibs`, `dev-lang/execline` and other `{sys-apps,net-dns,net-misc}/s6*` skarnet packages to the latest versions. I removed `/-fno-stack-protector$/d` from prepare step in all ebuilds because it was removed in upstream. Additionally, all ebuilds are fixed to respect `AR`, `CC` and `RANLIB`. I also dropped old versions, some of them are not compatible with currently present versions of theirs dependencies. And lastly, I appended myself to the metadata.xml because I would like to be proxy maintainer.

### CODE REVIEW

**Comment by arkamar** [comment]
I have git send-email him the patches directly.

**Comment by arkamar** [comment]
I pinged him.

**Comment by juippis** [comment]
@arkamar ping what's the status. Did williamh ever give feedback about these? I'd like to see the s6 stack bumped in Gentoo. Can you update these to latest version, and we'll take a look? 
Also ping @flexibeast if you have comments / want to take over.

**Comment by arkamar** [comment]
@thesamesam, @juippis guys, I made quite big revision of all related packages.

File: net-dns/s6-dns/s6-dns-2.3.5.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+inherit toolchain-funcs
+
+DESCRIPTION="Suite of DNS client programs and libraries for Unix systems"
+HOMEPAGE="https://www.skarnet.org/software/s6-dns/"
+SRC_URI="https://www.skarnet.org/software/${PN}/${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0/$(ver_cut 1-2)"
+KEYWORDS="~amd64 ~x86"
+IUSE="static static-libs"
```
Comment (juippis) [inline net-dns/s6-dns/s6-dns-2.3.5.3.ebuild:15]: I kind of fail to see how these two differ from each other.

File: net-dns/s6-dns/s6-dns-2.3.5.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		--bindir=/bin
+		--dynlibdir=/usr/$(get_libdir)
+		--libdir=/usr/$(get_libdir)/${PN}
+		--with-dynlib=/usr/$(get_libdir)
+		--with-lib=/usr/$(get_libdir)/skalibs
+		--with-sysdeps=/usr/$(get_libdir)/skalibs
+		--enable-shared
+		$(use_enable static allstatic)
+		$(use_enable static static-libc)
+		$(use_enable static-libs static)
```
Comment (juippis) [inline net-dns/s6-dns/s6-dns-2.3.5.3.ebuild:46]: Hilarious.

**Review by juippis — APPROVED** [review]
Newer versions build fine, but I'm having a lot of troubles with older versions:
```
/var/tmp/portage/vbslogs/execline-2.6.1.1-11.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/execline-2.6.1.1-11.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/execline-2.6.1.1-11.json:        "exit_code": 1,

src/libexecline/el_execsequence.c: In function 'el_execsequence':
src/libexecline/el_execsequence.c:32:3: error: implicit declaration of function 'xpathexec_r' [-    Werror=implicit-function-declaration[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index- Wimplicit-function-declaration]]
   32 |   xpathexec_r(argv2, envp, env_len(envp), fmt, j) ;
      |   ^~~~~~~~~~~
src/libexecline/el_execsequence.c:33:1: warning: 'noreturn' function does return
   33 | }
      | ^
cc1: some warnings being treated as errors
make: *** [Makefile:133: src/libexecline/el_execsequence.lo] Error 1
make: *** Waiting for unfinished jobs....



/var/tmp/portage/vbslogs/s6-portable-utils-2.2.3.0-8.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-portable-utils-2.2.3.0-8.json:        "exit_code": 1,

cc1: some warnings being treated as errors
make: *** [Makefile:130: src/skaembutils/s6-basename.o] Error 1
make: *** [Makefile:130: src/skaembutils/s6-env.o] Error 1
src/skaembutils/s6-cut.c: In function 'main':
src/skaembutils/s6-cut.c:151:5: error: unknown type name 'subgetopt_t'; did you mean 'subgetopt'?
  151 |     subgetopt_t l = SUBGETOPT_ZERO ;
      |     ^~~~~~~~~~~
      |     subgetopt

/usr/include/skalibs/sgetopt.h:22:65: note: expected 'subgetopt *' {aka 'struct subgetopt_s *'} but argument is of type 'int *'
   22 | extern int subgetopt_r (int, char const *const *, char const *, subgetopt *) ;
      |                                                                 ^~~~~~~~~~~
In file included from src/skaembutils/s6-head.c:9:
src/skaembutils/s6-head.c:111:36: error: request for member 'arg' in something not a structure or   union
  111 |           if (done || !uint0_scan(l.arg, &lines))
      |                                    ^
src/skaembutils/s6-head.c:119:36: error: request for member 'arg' in something not a structure or   union
  119 |           if (done || !uint0_scan(l.arg, &lines))
      |                                    ^
src/skaembutils/s6-head.c:128:14: error: request for member 'ind' in something not a structure or   union
  128 |     argc -= l.ind ; argv += l.ind ;
      |              ^
src/skaembutils/s6-head.c:128:30: error: request for member 'ind' in something not a structure or   union
  128 |     argc -= l.ind ; argv += l.ind ;
make: *** [Makefile:130: src/skaembutils/s6-head.o] Error 1
 * ERROR: sys-apps/s6-portable-utils-2.2.3.0::gentoo failed (compile phase):
 *   emake failed



/var/tmp/portage/vbslogs/s6-dns-2.3.3.0-12.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-dns-2.3.3.0-12.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-dns-2.3.3.0-12.json:        "exit_code": 1,

src/include/s6-dns/s6dns-engine.h:87:152: error: unknown type name 'tain_t'; did you mean 'tain'?
   87 | t *, uint32_t, char const *, unsigned int, uint16_t, s6dns_debughook_t const *, tain_t      const *, tain_t const *) ;
      |                                                                                 ^~~~~~
      |                                                                                 tain
src/include/s6-dns/s6dns-engine.h:87:168: error: unknown type name 'tain_t'; did you mean 'tain'?
   87 | har const *, unsigned int, uint16_t, s6dns_debughook_t const *, tain_t const *, tain_t      const *) ;
      |                                                                                 ^~~~~~
      |                                                                                 tain
src/include/s6-dns/s6dns-engine.h:93:64: error: unknown type name 'tain_t'; did you mean 'tain'?
   93 | extern void s6dns_engine_nextdeadline (s6dns_engine_t const *, tain_t *) ;
      |                                                                ^~~~~~
      |                                                                tain
src/include/s6-dns/s6dns-engine.h:100:52: error: unknown type name 'tain_t'; did you mean 'tain'?
  100 | extern int s6dns_engine_timeout (s6dns_engine_t *, tain_t const *) ;
      |                                                    ^~~~~~
      |                                                    tain
src/include/s6-dns/s6dns-engine.h:102:50: error: unknown type name 'tain_t'; did you mean 'tain'?
  102 | extern int s6dns_engine_event (s6dns_engine_t *, tain_t const *) ;
      |                                                  ^~~~~~
      |                                                  tain
make: *** [Makefile:133: src/libs6dns/s6dns_debughook_zero.lo] Error 1
make: *** Waiting for unfinished jobs....
 * ERROR: net-dns/s6-dns-2.3.3.0::gentoo failed (compile phase):
 *   emake failed



/var/tmp/portage/vbslogs/s6-rc-0.5.2.0-9.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-rc-0.5.2.0-9.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-rc-0.5.2.0-9.json:        "exit_code": 1,

cc1: some warnings being treated as errors
cc1: some warnings being treated as errors



/var/tmp/portage/vbslogs/s6-linux-utils-2.5.1.3-2.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-linux-utils-2.5.1.3-2.json:        "exit_code": 1,

cc1: some warnings being treated as errors



/var/tmp/portage/vbslogs/s6-networking-2.3.2.0-13.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-networking-2.3.2.0-13.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-networking-2.3.2.0-13.json:        "exit_code": 1,

cc1: some warnings being treated as errors



/var/tmp/portage/vbslogs/s6-2.9.2.0-4.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-2.9.2.0-4.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-2.9.2.0-4.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-2.9.2.0-4.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-2.9.2.0-4.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-2.9.2.0-4.json:        "exit_code": 1,

src/include/s6/ftrigr.h:57:3: error: unknown type name 'textclient_t'
   57 |   textclient_t connection ;
      |   ^~~~~~~~~~~~
In file included from src/libs6/ftrigr_subscribe.c:13:
src/include/s6/ftrigr.h:68:52: error: unknown type name 'tain_t'; did you mean 'tain'?
   68 | extern int ftrigr_start (ftrigr_t *, char const *, tain_t const *, tain_t *) ;
      |                                                    ^~~~~~
      |                                                    tain
In file included from src/libs6/ftrigr_unsubscribe.c:8:
src/include/s6/ftrigr.h:57:3: error: unknown type name 'textclient_t'
   57 |   textclient_t connection ;
      |   ^~~~~~~~~~~~
In file included from src/libs6/ftrigr_unsubscribe.c:8:
src/include/s6/ftrigr.h:68:52: error: unknown type name 'tain_t'; did you mean 'tain'?



/var/tmp/portage/vbslogs/s6-linux-init-1.0.4.2-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-linux-init-1.0.4.2-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-linux-init-1.0.4.2-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-linux-init-1.0.4.2-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-linux-init-1.0.4.2-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/s6-linux-init-1.0.4.2-0.json:        "exit_code": 1,

cc1: some warnings being treated as errors
```
I wouldn't mind merging them but doubt they'll build anymore on ~unstable anyway. 

FWIW LGTM. Will wait williamh's input for a day or two.

**Comment by flexibeast** [comment]
Re. the `tain_t` errors, cf. [this skalibs commit, "Huge incompatible changes"](http://git.skarnet.org/cgi-bin/cgit.cgi/skalibs/commit/src/include/skalibs/tai.h?id=9592bfd0dda7c575de07bce2c7a81b8432d845a4), which then appeared in [skalibs 2.11.0.0](http://git.skarnet.org/cgi-bin/cgit.cgi/skalibs/tag/?h=v2.11.0.0). So if you're bumping skalibs to 2.11.1.0, you'll need more recent versions of skalibs-dependent stuff, which will reflect the changes; otherwise you'll need to bump to a lower skalibs version.

**Comment by arkamar** [comment]
I will correct allowed dependencies in older versions later today or tomorrow.

**Comment by arkamar** [comment]
@juippis I am not sure if the last commit is ok according to gentoo standards.

**Comment by juippis** [comment]
Ideally you'd add those to the original commits where you modify these files and revbump. But I guess I'm not gonna ask you to do that. LGTM, s6-portable-utils-2.2.3.0 still dies to `cc1: some warnings being treated as errors`, `src/skaembutils/s6-dumpenv.c:25:47: warning: passing argument 4 of 'subgetopt_r' from incompatible  pointer type [-Wincompatible-pointer-types[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.      html#index-Wincompatible-pointer-types]]`

LGTM again, and thanks!

**Comment by arkamar** [comment]
Thanks guys! :)

**Pull request closed without merging**